frontend/src/app/components/crm/leads/leads.component.ts

```diff
@@ -149,6 +157,37 @@ export class LeadsComponent implements AfterViewInit {
     }
     return `${this.selection.isSelected(row) ? 'deselect' : 'select'} row ${row.leadId + 1}`;
   }
+  /*
+   * This method listens to the contact leads button
+   */
+  contactLead(){
+    // when no leads are selected
+    if(this.selection.selected.length == 0){
+      alert("Please select at least one lead.");
+      return;
+    }
+
+                                             //filter leads with no email
+    var recepients = this.selection.selected.filter(withEmail => withEmail.columnData.EMAIL != null)
```

Comment:
Should we be doing undefined checks here as well and what about if the property itself doesn't exist since I think it's an optional field.

Reply:
agree